src/sdl/surface.cpp

```diff
@@ -14,9 +14,90 @@
 #include "sdl/surface.hpp"
 
 #include "sdl/rect.hpp"
-#include "sdl/utils.hpp"
 #include "video.hpp"
 
+#include <iostream>
+
+const SDL_PixelFormat surface::neutral_pixel_format = []() {
+	SDL_PixelFormat format;
+
+#if SDL_VERSION_ATLEAST(2, 0, 6)
+	surface surf(
+		SDL_CreateRGBSurfaceWithFormat(0, 1, 1, 32, SDL_PIXELFORMAT_ARGB8888));
+#else
+	surface surf(
+		SDL_CreateRGBSurface(0, 1, 1, 32, SDL_RED_MASK, SDL_GREEN_MASK, SDL_BLUE_MASK, SDL_ALPHA_MASK));
+#endif
+
+	format = *surf->format;
+	format.palette = nullptr;
+
+	return format;
+}();
+
+surface::surface(SDL_Surface* surf)
+	: surface_(surf)
+{
+	make_neutral(); // EXTREMELY IMPORTANT!
+}
+
+surface::surface(int w, int h)
+	: surface_(nullptr)
+{
+	if (w < 0 || h < 0) {
+		std::cerr << "error: creating surface with negative dimensions\n";
+		return;
```

Comment:
Are you sure you don't want to throw here? Now there is a constructor that can return an incorrectly initialized surface.

Reply:
I was considering also taking the absolute value. Do you think throwing is better?

Reply:
Is there ever going to be a case where using the absolute value is the intended behaviour? It seems to me like negative width or height can only be a bug that should not be hidden.

Reply:
`throw` added in ef47599d966b82271b52ce65061918d35b782814.